tests/api.c

```diff
@@ -2497,14 +2497,13 @@ static int test_wc_UpdateMd5 (void)
         return SSL_FAILURE;
     }
 
-    /* If not returned then the unit test passed test vectors. */
-    printf(resultFmt, 0 == 0 ? passed : failed);
+    printf(resultFmt, ret == BAD_FUNC_ARG ? passed : failed);
```

Comment:
Although this now correctly prints "failed" if this function fails, it appears that "tests/unit.test" still completes successfully.  To test this, you can explicitly force a failure by returning a negative value after one of the calls in this function.

This applies to other hash tests as well.